connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -197,19 +197,19 @@ private[sql] object AvroUtils extends Logging {
 
     def hasNextRow: Boolean = {
       while (!completed && currentRow.isEmpty) {
-        if (fileReader.pastSync(stopPosition)) {
+        // In some cases of empty blocks in an Avro file, `fileReader.hasNext()` returns false but
+        // advances the cursor to the next block, so we need to call `fileReader.hasNext()` again
+        // to correctly report whether the next record exists.
+        val r = (fileReader.hasNext || fileReader.hasNext) && !fileReader.pastSync(stopPosition)
```

Comment:
```suggestion
        val noMoreData = (fileReader.hasNext || fileReader.hasNext) && !fileReader.pastSync(stopPosition)
```

Reply:
Thanks. I updated the code but I used `moreData` variable instead and `if (!moreData)`.